### PR TITLE
[mjit] Invoke mini through ICompiler

### PR DESCRIPTION
**Pull request by luhenry**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -40,7 +40,7 @@ MONO_VERSION_BUILD=`echo $VERSION | cut -d . -f 3`
 # This can be reset to 0 when Mono's version number is bumped
 # since it's part of the corlib version (the prefix '1' in the full
 # version number is to ensure the number isn't treated as octal in C)
-MONO_CORLIB_COUNTER=25
+MONO_CORLIB_COUNTER=26
```
Comment (lewurm) [inline configure.ac]: why is this needed?

**Comment by lewurm** [comment]
awesome 🙂 just to make sure before merging: we can still normally build mono with your changes, right? MJIT will only be used if `MONO_MJIT` is set if I understand correctly?

File: configure.ac (hunk 1 above)
Comment (luhenry) [reply configure.ac]: You need to make sure to have the latest monolite, otherwise you might be missing `Mono.Compiler`
